core/capabilities/ccip/ccipevm/msghasher.go

```diff
@@ -113,7 +114,7 @@ func (h *MessageHasherV1) Hash(ctx context.Context, msg cciptypes.Message) (ccip
 		lggr.Debugw("decoded dest gas amount",
 			"destGasAmount", destGasAmount)
 
-		destTokenAddress, err := abiDecodeAddress(rta.DestTokenAddress)
+		destTokenAddress, err := abiDecodeAddress(common.LeftPadBytes(rta.DestTokenAddress, 32))
```

Comment:
I feel like this may end up masking some issues - the contracts have to emit something that must be abi-encoded if the destination is EVM.

Reply:
how about use `common.BytesToAddress(tokenAmount.SourcePoolAddress)`, same with the [executecodec](https://github.com/smartcontractkit/chainlink/blob/a9ccdda94a09549c8d720fa07828603bab3855b7/core/capabilities/ccip/ccipevm/executecodec.go#L99)

Reply:
> Or do we expect the onchain user on source chain to provide an ABI encoded []byte for destination addresses?

The onchain user _must_ specify a valid receiver address for the dest chain family. This is enforced on EVM -> EVM and EVM -> SOL, should be enforced the other way around as well.

Reply:
If we don't have this kind of behavior & validation in solana then I believe this is a bug and should be fixed?

Reply:
@makramkd : How could we validate on source chain, whether destination address is valid?

Reply:
We specify a valid address, but we specify it as 20 bytes, not the fully padded 32 bytes

Reply:
I have the same question as @prashantkumar1982 and @huangzhen1997

It seems like this needs some followup to make sure this handles all families with the address codec.

Could we add a TODO or create a followup ticket?